pallets/tasks/src/lib.rs

```diff
@@ -260,6 +265,10 @@ pub mod pallet {
 	#[pallet::storage]
 	pub type BatchTaskId<T: Config> = StorageMap<_, Blake2_128Concat, BatchId, TaskId, OptionQuery>;
 
+	/// List of failed batches.
+	#[pallet::storage]
+	pub type FailedBatchIds<T: Config> = StorageValue<_, Vec<BatchId>, ValueQuery>;
```

Comment:
```suggestion
	pub type FailedBatchIds<T: Config> = StorageMap<_, BatchId, (), ValueQuery>;
```

Using a HashSet is way more efficient than using a vec...